### PR TITLE
app/vmctl: enable range steps in reverse order

### PR DESCRIPTION
**Pull request by dmitryk-dk**
Enable reverse mode for range steps. In that mode all ranges will be sorted from newest date to oldest. 

Related issue: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5376

Signed-off-by: dmitryk-dk [d.kozlov@victoriametrics.com](mailto:d.kozlov@victoriametrics.com)

### CODE REVIEW

File: app/vmctl/flags.go [hunk 1]
```diff
@@ -320,10 +320,11 @@ var (
...
 
 const (
-	vmNativeFilterMatch     = "vm-native-filter-match"
-	vmNativeFilterTimeStart = "vm-native-filter-time-start"
-	vmNativeFilterTimeEnd   = "vm-native-filter-time-end"
-	vmNativeStepInterval    = "vm-native-step-interval"
+	vmNativeFilterMatch             = "vm-native-filter-match"
+	vmNativeFilterTimeStart         = "vm-native-filter-time-start"
+	vmNativeFilterTimeEnd           = "vm-native-filter-time-end"
+	vmNativeFilterTimeReverseChunks = "vm-native-filter-time-reverse-chunks"
```
Comment (hagen1778) [inline app/vmctl/flags.go:326]: why you call it "chunks"? could it be just `vm-native-filter-time-reverse`?

File: app/vmctl/flags.go [hunk 2]
```diff
@@ -366,6 +367,11 @@ var (
 				stepper.StepMonth, stepper.StepWeek, stepper.StepDay, stepper.StepHour, stepper.StepMinute),
 			Value: stepper.StepMonth,
 		},
+		&cli.BoolFlag{
+			Name:  vmNativeFilterTimeReverseChunks,
+			Usage: "Sort time chunks in reverse mode. In that mode time ranges will be sorted from newest to oldest time ranges.",
```
Comment (hagen1778) [inline app/vmctl/flags.go:372]:
```suggestion
			Usage: "Whether to reverse order of time intervals split by '-vm-native-step-interval' cmd-line flag. When set, the migration will start from the newest to the oldest intervals.",
```

File: app/vmctl/flags.go (hunk 1 above)
Comment (dmitryk-dk) [reply app/vmctl/flags.go]: renamed this flag

File: app/vmctl/flags.go (hunk 2 above)
Comment (dmitryk-dk) [reply app/vmctl/flags.go]: fixed

File: app/vmctl/stepper/split.go [hunk 3]
```diff
@@ -71,6 +72,12 @@ func SplitDateRange(start, end time.Time, step string) ([][]time.Time, error) {
 		ranges = append(ranges, []time.Time{s, e})
 		currentStep = e
 	}
+	if timeReverse {
+		sort.SliceStable(ranges, func(i, j int) bool {
+			return ranges[i][0].After(ranges[j][0])
+		})
+		return ranges, nil
```
Comment (zekker6) [inline app/vmctl/stepper/split.go:79]: I don't see why early return is needed here, can we remove it?

File: app/vmctl/flags.go [hunk 4]
```diff
@@ -524,6 +531,11 @@ var (
 			Usage:    fmt.Sprintf("Split export data into chunks. Requires setting --%s. Valid values are %q,%q,%q,%q.", remoteReadFilterTimeStart, stepper.StepMonth, stepper.StepDay, stepper.StepHour, stepper.StepMinute),
```
Comment (zekker6) [inline app/vmctl/flags.go:531]: Wdyt about adding a comment for this flag which will mention that by default migration order is from oldest to newest intervals and reference to new flag to change this behaviour?

Comment (dmitryk-dk) [reply app/vmctl/flags.go]: comment was added

**Review by zekker6 — APPROVED** [review]

**Review by hagen1778 — APPROVED** [review]
Thanks!

**Pull request merged**